proof-of-spacetime.md

```diff
@@ -0,0 +1,197 @@
+## Proof-of-Spacetime
+
+This document describes Rational-PoSt, the Proof-of-Spacetime used in Filecoin.
+
+## Rational PoSt
+
+### Definitions
+
+| Name | Value |Description |
+|------|-------|------------|
+| `POST_PROVING_PERIOD` | `2880` blocks  (~24h) | The time interval in which a PoSt has to be submitted. |
+| `POST_CHALLENGE_TIME` | `240` blocks (~2h) | The time offset at which the actual work of generating the PoSt should be started. This is some delta before the end of the `Proving Period`, and as such less then a single `Proving Period`. |
+
+{{% notice todo %}}
+**TODO**: The above values are tentative and need both backing from research as well as detailed reasoning why we picked them.
+{{% /notice %}}
+
+### High Level API
+
+#### Fault Detection
+
+Fault detection happens over the course of the life time of a sector. When the sector is for some reason unavailable, the miner is responsible to submit the known `faults` using the `AddFaults` message to the chain.
+Only faults which have been reported at challenge time, will be accounted for. If any other faults have occured the miner can not submit a valid PoSt for this proving period.
+
+The PoSt generation then takes the latest available `faults` of the miner to generate a PoSt matching the committed sectors and faults.
+
+At the beginning of a new proving period all faults are reset, and if they persist the miner needs to resubmit an `AddFaults` message.
+
+If the miner knows that the sectors are permanently lost, they can submit them as part of the `doneSet`, to ensure they are removed from the proving set.
+
+
+{{% notice todo %}}
+**TODO**: The penalization for faults is not clear yet.
+{{% /notice %}}
+
+#### Generation
+
+`GeneratePoSt` generates a __*Proof of Spacetime*__ over all  __*sealed sectors*__ of a single miner— identified by their `commR` commitments. This is accomplished by performing a series of merkle inclusion proofs (__*Proofs of Retrievability*__). Each proof is of a challenged node in a challenged sector. The challenges are generated pseudo-randomly, based on the provided `seed`. At each time step, a number of __*Proofs of Retrievability*__ are performed.
+
+```go
+// Generate a new PoSt.
+func GeneratePoSt(sectorSize BytesAmount, sectors SectorSet, seed Seed, faults FaultSet) PoStProof {
+    // Generate the Merkle Inclusion Proofs + Faults
+
+    inclusionProofs := []
+	sectorsSorted := []
+    challenges := DerivePoStChallenges(seed, faults, sectorSize, len(sectors))
+
+    for i := 0; i < len(challenges); i++ {
+        challenge := challenges[i]
+
+        // Leaf index of the selected sector
+        inclusionProof, isFault := GenerateMerkleInclusionProof(challenge.Sector, challenge.Leaf)
+        if isFault {
+            // faulty sector, need to post a fault to the chain and try to recover from it
```

Comment:
Comment should be more like:

"Detected late fault: The data needed for this post submission is missing, report either as a fault (if you are planning to recover it) and also as a done the missing sector (if you cannot recover)."